plugins/janus_audiobridge.c

```diff
@@ -2846,10 +2874,79 @@ void janus_audiobridge_incoming_rtp(janus_plugin_session *handle, int video, cha
 			g_free(pkt);
 			return;
 		}
-		pkt->length = opus_decode(participant->decoder, payload, plen, (opus_int16 *)pkt->data, BUFFER_SAMPLES, USE_FEC);
-		g_atomic_int_set(&participant->decoding, 0);
+
+		/* Check sequence number received, verify if relevant to expected one. */
+		if(pkt->seq_number == participant->expected_seq) {
+			/* regular decode */
+			pkt->length = opus_decode(participant->decoder, payload, plen, (opus_int16 *)pkt->data, BUFFER_SAMPLES, 0);
+
+			/* update last_timestamp */
+			participant->last_timestamp = pkt->timestamp;
+
+			/* Increment according to previous seq_number */
+			participant->expected_seq = pkt->seq_number + 1;
+		} else if(pkt->seq_number > participant->expected_seq) {
```

Comment:
Leaving this as a reminder for the next days.
This comparison is not dealing with uint16 wraparound, leading to unexpected behavior in case of misordered packets occurring near the uint16 upper limit (e.g. expected seq 2, seq number 65533). In particular the expected seq number will be wrongly updated a few lines below.